src/System.CommandLine/Parsing/ArgumentResult.cs

```diff
@@ -55,115 +53,152 @@ public sealed class ArgumentResult : SymbolResult
         /// <param name="numberOfTokens">The number of tokens to take. The rest are passed on.</param>
         /// <exception cref="ArgumentOutOfRangeException">numberOfTokens - Value must be at least 1.</exception>
         /// <exception cref="InvalidOperationException">Thrown if this method is called more than once.</exception>
+        /// <exception cref="NotSupportedException">Thrown if this method is called by Option-owned ArgumentResult.</exception>
         public void OnlyTake(int numberOfTokens)
         {
             if (numberOfTokens < 0)
             {
                 throw new ArgumentOutOfRangeException(nameof(numberOfTokens), numberOfTokens, "Value must be at least 1.");
             }
 
-            if (PassedOnTokens is { })
+            if (_passedOnHasBeenCalled)
             {
                 throw new InvalidOperationException($"{nameof(OnlyTake)} can only be called once.");
             }
 
-            if (_tokens is not null)
+            if (Parent is OptionResult)
             {
-                var passedOnTokensCount = _tokens.Count - numberOfTokens;
+                throw new NotSupportedException($"TakeOnly is supported only by Command-owned ArgumentResults");
```

Comment:
```suggestion
                throw new NotSupportedException($"{nameof(OnlyTake)} is supported only for a {nameof(Command)}-owned {nameof(ArgumentResult)}");
```

...mainly because at least one of these is likely to get renamed.